### PR TITLE
bpo-34603: Fix returning structs from functions

### PR DESCRIPTION
**Pull request by vladima**
Per [MSDN: Return values](https://msdn.microsoft.com/en-us/library/7572ztz4.aspx):
>To be returned by value in RAX, user-defined types must have a length of 1, 2, 4, 8, 16, 32, or 64 bits

Effectively if size of struct is not 1, 2, 4 or 8 bytes - caller needs to allocate a memory for the result and pass a pointer as a first argument.

https://bugs.python.org/issue34603

### CODE REVIEW

File: Lib/ctypes/test/test_win32.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def test_noargs(self):
 class ReturnStructSizesTestCase(unittest.TestCase):
     def test_sizes(self):
         dll = CDLL(_ctypes_test.__file__)
-        for i in range(3, 9):
+        for i in range(1, 9):
```
Comment (vstinner) [inline Lib/ctypes/test/test_win32.py]: Why not testing up to 20 bytes, just in case?

**Review by vstinner — APPROVED** [review]
LGTM.

Oh, I'm sorry, I didn't notice that each additional tested size requires to write a long C code in _ctypes_test.c. IMHO it's not worth it and adds too much maintenance burden. I was concerned about a special case for size=16 bytes, but again, I don't think that it's worth it.

Would you mind to remove tests for 11..20, so only test 1..10? So we cover all special cases on 64-bit + size 9 and size 10 -- just in case.

**Comment by vstinner** [comment]
@vladima: Would you mind to check if the issue exists upstream? https://github.com/libffi/libffi/tree/master/src

**Comment by vladima** [comment]
@vstinner code upstream is a bit reshaped but at glance this issue is already fixed.

**Comment by vstinner** [comment]
> @vstinner code upstream is a bit reshaped but at glance this issue is already fixed.

Cool, thanks for checking!

**Pull request merged**